### PR TITLE
feat: change the `Artifact` to be folder-based

### PR DESCRIPTION
**Pull request by yzld2002**
Currently we implement the `Artifact` based on the pickle.
This requires to load / save the files.

This PR changes the `Artifact` to be folder-based.
There is no need to load / save any more.
All the files under the `cwd` (current working directory) will be treated as the files inside the Artifact.

Also we want to keep the `chat_with_artifacts` running in the same environment with the code_interpreter. So we don't have to maintain "local" and "remote".

---

Local test

Code:

```python
from vision_agent.agent import VisionAgent
from vision_agent.tools.meta_tools import Artifacts


artifacts = Artifacts("/Users/zhichao/Projects/vision-agent-chat/va-local-test")

agent = VisionAgent(verbosity=1)
resp, artifacts = agent.chat_with_artifacts(
    [
        {
            "role": "user",
            "content": "Can you write code to detect the car and visualize the detection?",
        }
    ],
    artifacts=artifacts,
    test_multi_plan=False,
)
resp.append({"role": "user", "content": "Can you describe the output image?"})
resp, artifacts = agent.chat_with_artifacts(resp, artifacts, test_multi_plan=False)
```

Result:
image

image

---

Also tested the `view_media_artifact` action, the result looks good now.
image

### CODE REVIEW

File: vision_agent/agent/vision_agent.py [hunk 1]
```diff
@@ -397,41 +379,15 @@ def chat_with_artifacts(
...
-            for chat_i in int_chat:
-                if "media" in chat_i:
-                    for media in chat_i["media"]:
-                        media = cast(str, media)
-                        artifacts.artifacts[Path(media).name] = open(media, "rb").read()
-
-                        media_remote_path = (
-                            Path(artifacts.remote_save_path.parent) / Path(media).name
-                        )
-                        chat_i["content"] += f" Media name {media_remote_path}"  # type: ignore
```
Comment (dillonalaird) [inline vision_agent/agent/vision_agent.py:432]: This line needs to stay in here, or at least somehow attach this to the message content. This is so the agent knows where the media path is so when it executes code it loads the correct path. You can see it in the prompt examples here https://github.com/landing-ai/vision-agent/blob/main/vision_agent/agent/vision_agent_prompts.py#L43

**Review by dillonalaird — CHANGES_REQUESTED** [review]
Thanks for doing this! I think it mostly looks good, just one comment right now

File: vision_agent/agent/vision_agent.py [hunk 2]
```diff
@@ -553,14 +500,6 @@ def chat_with_artifacts(
...
                     )
                     obs_chat_elt: Message = {"role": "observation", "content": obs}
-                    media_obs = check_and_load_image(code_action)
-                    if media_obs and result.success:
-                        # media paths will be under the local_save_path when we download
-                        # them after each turn
-                        obs_chat_elt["media"] = [
-                            artifacts.local_save_path.parent / media_ob
-                            for media_ob in media_obs
-                        ]
```
Comment (dillonalaird) [inline vision_agent/agent/vision_agent.py:563]: Here if the agent calls `view_media_artifact`, we then add that to the media key of the current chat so the agent can actually see the image. I don't see this case being handled anywhere with these lines deleted

**Review by dillonalaird — CHANGES_REQUESTED** [review]
one more comment

File: vision_agent/agent/vision_agent.py (hunk 1 above)
Comment (yzld2002) [reply vision_agent/agent/vision_agent.py:432]: addressed

File: vision_agent/agent/vision_agent.py [hunk 3]
```diff
@@ -554,12 +509,10 @@ def chat_with_artifacts(
                     )
                     obs_chat_elt: Message = {"role": "observation", "content": obs}
                     media_obs = check_and_load_image(code_action)
+                    print(media_obs)
```
Comment (dillonalaird) [inline vision_agent/agent/vision_agent.py:512]: NIT probably want to remove print

**Review by dillonalaird — APPROVED** [review]
Looks good! One minor comment

File: vision_agent/agent/vision_agent.py (hunk 3 above)
Comment (yzld2002) [reply vision_agent/agent/vision_agent.py]: removed

**Pull request merged**